src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -74,6 +96,11 @@ public sealed override FieldSymbol TupleUnderlyingField
         {
             get
             {
+                if (!TypeSymbol.Equals(_underlyingField.ContainingType, _containingTuple.TupleUnderlyingType, TypeCompareKind.ConsiderEverything))
+                {
+                    return null;
+                }
+
                 return _containingTuple.GetTupleMemberSymbolForUnderlyingMember(_underlyingField.AssociatedSymbol);
```

Comment:
>AssociatedSymbol [](start = 97, length = 16)

When is there an associated symbol for a tuple field?

Reply:
I think you're right. This can be simplified (always null). Will add test explicitly for this scenario.

---
In reply to: [597362368](https://github.com/dotnet/roslyn/pull/44231#discussion_r597362368) [](ancestors = 597362368)